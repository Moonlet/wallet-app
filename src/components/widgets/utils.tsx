import React from 'react';
import { DataType, ICurrencyData, IData, IDataStyle, ITextData } from './types';
import { Text } from '../../library';
import BigNumber from 'bignumber.js';
import { formatNumber } from '../../core/utils/format-number';
import { normalize, normalizeFontAndLineHeight } from '../../styles/dimensions';

const formatTextData = (data: ITextData): string => {
    let text = '';
    // Static text
    text = String(data.value);

    if (data?.params) {
        // TODO
    }

    // Translated text
    // TODO

    return text;
};

const amountFromStd = (value: BigNumber | number | string, decimals: number): BigNumber => {
    return new BigNumber(value).dividedBy(new BigNumber(10).pow(decimals));
};

const formatCurrencyData = (data: ICurrencyData): string => {
    let text = '';
    let value = data.value;

    if (data?.decimals) {
        value = amountFromStd(value, data.decimals).toFixed();
    }

    text = formatNumber(Number(value), {
        currency: data.symbol,
        maximumFractionDigits: data?.round?.decimals
    });

    if (data?.beautify) {
        const out = beautify(value);
        text = new BigNumber(out.value).toFixed(data.beautify.decimals) + out.unit;
    }

    return text;
};

const formatStyles = (style: IDataStyle): { [key: string]: string | number } => {
    const finalStyle = {};

    for (const s of Object.keys(style)) {
        const sVal: any = style[s];

        if (typeof sVal === 'object') {
            if (sVal.fn === 'normalize') {
                finalStyle[s] = normalize(sVal.value);
            }
            if (sVal.fn === 'normalizeFontAndLineHeight') {
                finalStyle[s] = normalizeFontAndLineHeight(sVal.value);
            }
        } else {
            finalStyle[s] = style[s];
        }
    }

    return finalStyle;
};

const beautify = (
    value: BigNumber | string | number,
    fromValue?: number
): {
    value: BigNumber | string;
    unit: string;
} => {
    fromValue = fromValue || 1000;

    value = new BigNumber(value);

    let out = { value, unit: '' };

    if (value.gte(fromValue)) {
        const units = [
            [1000000000, 'B'],
            [1000000, 'M'],
            [1000, 'k']
        ];

        for (const unit of units) {
            if (value.gte(unit[0])) {
                out = {
                    value: value.div(unit[0]),
                    unit: unit[1].toString()
                };
                break;
            }
        }
    }

    return out;
};

export const formatDataJSXElements = (data: IData | IData[], baseStyle: any): JSX.Element[] => {
    if (!Array.isArray(data)) {
        data = [data];
    }

    const finalData: JSX.Element[] = [];

    for (const d of data) {
        switch (d.type) {
            case DataType.TEXT:
                finalData.push(
                    <Text style={[baseStyle, d?.style && formatStyles(d.style)]}>
                        {formatTextData(d.data as ITextData)}
                    </Text>
                );
                break;

            case DataType.CURRENCY:
                finalData.push(
                    <Text style={[baseStyle, d?.style && formatStyles(d.style)]}>
                        {formatCurrencyData(d.data as ICurrencyData)}
                    </Text>
                );
                break;

            default:
                break;
        }
    }

    return finalData;
};
