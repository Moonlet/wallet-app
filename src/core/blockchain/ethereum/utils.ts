import { TransactionStatus } from '../../wallet/types';

export const getEthStatusByCode = (status): TransactionStatus => {
    switch (parseInt(status, 16)) {
        case 0:
            return TransactionStatus.FAILED;
        case 1:
            return TransactionStatus.SUCCESS;
        case 2:
            return TransactionStatus.PENDING;
        default:
            return TransactionStatus.FAILED;
    }
};
