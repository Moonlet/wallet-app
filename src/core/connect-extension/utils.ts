import { getBlockchain } from '../blockchain/blockchain-factory';
import { NavigationParams } from 'react-navigation';
import { NavigationService } from '../../navigation/navigation-service';

export const getUrlParams = (search = ``) => {
    const hashes = search.slice(search.indexOf('?') + 1).split('&');
    return hashes.reduce((params, hash) => {
        const split = hash.indexOf('=');

        if (split < 0) {
            return { ...params, [hash]: null };
        }

        const key = hash.slice(0, split);
        const val = hash.slice(split + 1);

        return { ...params, [key]: decodeURIComponent(val) };
    }, {});
};

export const signExtensionTransaction = payload => {
    const data = payload.params[0];
    const blockchainConfig = getBlockchain(data.selectedAccount.blockchain).config;
    const navigationParams: NavigationParams = {
        blockchain: data.selectedAccount.blockchain,
        accountIndex: data.selectedAccount.index,
        token: data.token,
        tokenLogo: blockchainConfig.tokens[blockchainConfig.coin].icon,
        extensionTransactionPayload: payload
    };

    if (NavigationService.getCurrentRoute() === 'Token') {
        NavigationService.replace('Token', navigationParams);
    } else {
        NavigationService.navigate('Token', navigationParams);
    }
};
