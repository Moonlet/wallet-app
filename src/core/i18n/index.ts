export * from './translation/translate-component';
export * from './translation/translate';
