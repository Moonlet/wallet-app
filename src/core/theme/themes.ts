export enum Theme {
    dark = 'dark',
    light = 'light'
}
