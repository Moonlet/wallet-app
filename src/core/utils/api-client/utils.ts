import moment from 'moment';
import CONFIG from '../../../config';
import * as schnorr from '@zilliqa-js/crypto/dist/schnorr';

// Refactor this using ntp time
export const getTimestamp = () => {
    return moment().unix() * 1000;
};

export const getDomain = () => {
    return new URL(CONFIG.walletApiBaseUrl).host;
};

export const getSignature = (
    data: any,
    walletPrivateKey: string,
    walletPublicKey: string
): string => {
    const sig = schnorr.sign(
        Buffer.from(JSON.stringify(data), 'hex'),
        Buffer.from(walletPrivateKey, 'hex'),
        Buffer.from(walletPublicKey, 'hex')
    );

    return sig.r.toString('hex') + sig.s.toString('hex');
};
