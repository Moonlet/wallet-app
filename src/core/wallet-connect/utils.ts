import { NavigationService } from '../../navigation/navigation-service';
import { getBlockchain } from '../blockchain/blockchain-factory';
import { NavigationParams } from 'react-navigation';

export const signExtensionTransaction = payload => {
    const data = payload.params[0];
    const blockchainConfig = getBlockchain(data.selectedAccount.blockchain).config;
    const navigationParams: NavigationParams = {
        blockchain: data.selectedAccount.blockchain,
        accountIndex: data.selectedAccount.index,
        token: data.token,
        tokenLogo: blockchainConfig.tokens[blockchainConfig.coin].icon,
        extensionTransactionPayload: payload
    };

    if (NavigationService.getCurrentRoute() === 'Token') {
        NavigationService.replace('Token', navigationParams);
    } else {
        NavigationService.navigate('Token', navigationParams);
    }
};
