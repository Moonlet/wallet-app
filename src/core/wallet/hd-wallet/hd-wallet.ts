import { IWallet } from '../types';
import { Mnemonic } from './mnemonic';
import { Blockchain, IBlockchainTransaction, DerivationType } from '../../blockchain/types';
import { getBlockchain } from '../../blockchain/blockchain-factory';
import { IAccountState } from '../../../redux/wallets/state';
import { HDKeyFactory } from './hd-key/hdkey-factory';
import { readEncrypted } from '../../secure/storage/storage';
import { getEncryptionKey } from '../../secure/keychain/keychain';

export class HDWallet implements IWallet {
    public static async loadFromStorage(walletId: string, pass: string): Promise<HDWallet> {
        const encryptionKey = await getEncryptionKey(pass);
        const data = await readEncrypted(walletId, encryptionKey);
        return Promise.resolve(new HDWallet(data.toString()));
    }
    private mnemonic: string;
    private seedCache: Buffer;

    constructor(mnemonic: string) {
        if (Mnemonic.verify(mnemonic)) {
            this.mnemonic = mnemonic;
        } else {
            throw new Error('Invalid mnemonic.');
        }
    }

    get seed() {
        if (!this.seedCache) {
            this.seedCache = Mnemonic.toSeed(this.mnemonic);
        }

        return this.seedCache;
    }

    public getAccounts(
        blockchain: Blockchain,
        index: number,
        indexTo?: number
    ): Promise<IAccountState[]> {
        indexTo = indexTo || index;

        if (isNaN(Number(index))) {
            return Promise.reject(
                `${this.constructor.name}.getAccounts(): index must be a positive number.`
            );
        } else if (index < 0) {
            return Promise.reject(
                `${this.constructor.name}.getAccounts(): index must be a positive number.`
            );
        }

        if (isNaN(Number(indexTo))) {
            return Promise.reject(
                `${this.constructor.name}.getAccounts(): indexTo must be a positive number.`
            );
        }

        if (indexTo < index) {
            return Promise.reject(
                `${this.constructor.name}.getAccounts(): indexTo value must be greated than index value.`
            );
        }

        try {
            const accounts = [];
            const blockchainInstance = getBlockchain(blockchain);
            const key = HDKeyFactory.get(
                blockchainInstance.config.derivationType,
                this.seed
            ).derive(blockchainInstance.config.derivationPath);
            for (let i = index; i <= indexTo; i++) {
                const accountDerivationPath = blockchainInstance.account.getAccountDerivationPath(
                    i
                );
                const derivation = key.derive(`m/${accountDerivationPath}`);
                const privateKey = blockchainInstance.account.getPrivateKeyFromDerived(derivation);
                accounts.push(blockchainInstance.account.getAccountFromPrivateKey(privateKey, i));
            }
            return Promise.resolve(accounts);
        } catch (e) {
            if (e.message.indexOf('implementation not found')) {
                return Promise.reject(`Blockchain implementation not found.`);
            } else {
                return Promise.reject('There was an error while generating the accounts.');
            }
        }
    }

    public getPrivateKey(blockchain: Blockchain, accountIndex: number): string {
        const blockchainInstance = getBlockchain(blockchain);
        const key = HDKeyFactory.get(blockchainInstance.config.derivationType, this.seed).derive(
            blockchainInstance.config.derivationPath
        );
        const derivation = key.derive(
            `m/${blockchainInstance.account.getAccountDerivationPath(accountIndex)}`
        );
        return blockchainInstance.account.getPrivateKeyFromDerived(derivation);
    }

    public async sign(
        blockchain: Blockchain,
        accountIndex: number,
        tx: IBlockchainTransaction
    ): Promise<any> {
        return getBlockchain(blockchain).transaction.sign(
            tx,
            this.getPrivateKey(blockchain, accountIndex)
        );
    }

    public getMnemonic() {
        return this.mnemonic;
    }

    public getWalletCredentials(): { publicKey: string; privateKey: string } {
        const key = HDKeyFactory.get(DerivationType.HD_KEY, this.seed).derive(
            `m/${'moonlet'
                .split('')
                .map(l => l.charCodeAt(0))
                .join('/')}`
        );

        const blockchainInstance = getBlockchain(Blockchain.ZILLIQA);
        const privateKey = blockchainInstance.account.getPrivateKeyFromDerived(key);
        const publicKey = blockchainInstance.account.privateToPublic(privateKey);

        return {
            publicKey,
            privateKey
        };
    }
}
