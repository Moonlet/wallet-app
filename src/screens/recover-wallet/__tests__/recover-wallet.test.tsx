import React from 'react';
import {
    RecoverWalletScreenComponent,
    IProps,
    navigationOptions,
    IReduxProps
} from '../recover-wallet';
import { darkTheme } from '../../../styles/themes/dark-theme';
import styleProvider from '../styles';

import { shallow } from 'enzyme';
import { loadTranslations } from '../../../core/i18n';
import { Mnemonic } from '../../../core/wallet/hd-wallet/mnemonic';
import { testUtils } from '../../../core/utils/test';

const props: IProps & IReduxProps = {
    // @ts-ignore
    navigation: {
        navigate: jest.fn()
    },
    styles: styleProvider(darkTheme),
    tosVersion: 1,
    createHDWallet: jest.fn((mnemonic, callback): any => callback()),
    theme: darkTheme
};

jest.mock('../../../core/wallet/hd-wallet/mnemonic');
jest.mock('../../../core/constants/app', () => {
    return {
        TOS_VERSION: 1
    };
});

const validMnemonic =
    'panic club above clarify orbit resist illegal feel bus remember aspect field test bubble dog trap awesome hand room rice heavy idle faint salmon';

describe('creat wallet terms screen component', () => {
    beforeAll(async () => {
        // Mnemonic.generate = jest.fn(() =>
        //     Promise.resolve('1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24')
        // );

        await loadTranslations('en');
    });

    test('renders correctly', async () => {
        const wrapper = shallow(<RecoverWalletScreenComponent {...props} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });

    test('navigates to tos when there are new tos', () => {
        const wrapper = shallow(<RecoverWalletScreenComponent {...{ ...props, tosVersion: 0 }} />);
        expect(props.navigation.navigate).toHaveBeenCalledWith('CreateWalletTerms');
    });

    test('sets correct navigation options', () => {
        const navigationProp = { navigation: { state: { params: { goBack: jest.fn() } } } };
        const options = navigationOptions(navigationProp);
        expect(options.headerLeft()).toMatchSnapshot();
        expect(options).toMatchSnapshot();
    });

    test('does not have a back button if no goBack param is set', () => {
        const navigationProp = { navigation: {} };
        const options = navigationOptions(navigationProp);
        expect(options.headerLeft()).toBe(null);
    });

    test('fills first input corectly', () => {
        const wrapper = shallow(<RecoverWalletScreenComponent {...props} />);

        // doesnt work
        wrapper.find('[testID="button-next"]').simulate('press');
        wrapper.find('[testID="button-paste"]').simulate('press');
        expect(wrapper).toMatchSnapshot();
    });

    test('generates suggestions correctly and sets correct mnemonic based on suggestion', () => {
        const spySetState = jest.spyOn(RecoverWalletScreenComponent.prototype, 'setState');
        const wrapper = shallow(<RecoverWalletScreenComponent {...props} />);
        const instance = wrapper.instance();

        // @ts-ignore
        instance.setMnemonicText(0, 'pa');
        expect(spySetState).toBeCalledWith({
            indexForSuggestions: 0,
            mnemonic: [
                'pa',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                ''
            ],
            suggestions: [
                'pact',
                'paddle',
                'page',
                'pair',
                'palace',
                'palm',
                'panda',
                'panel',
                'panic',
                'panther',
                'paper',
                'parade',
                'parent',
                'park',
                'parrot',
                'party',
                'pass',
                'patch',
                'path',
                'patient',
                'patrol',
                'pattern',
                'pause',
                'pave',
                'payment'
            ]
        });

        wrapper.find('[testID="button-suggestion-0"]').simulate('press');

        expect(spySetState).toBeCalledWith({
            mnemonic: [
                'pact',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                '',
                ''
            ]
        });
    });

    test('marks errors', () => {
        const wrapper = shallow(<RecoverWalletScreenComponent {...props} />);
        wrapper.find('[testID="button-confirm"]').simulate('press');
        expect(wrapper).toMatchSnapshot();
    });

    test('creates wallet and navigates to dashboard', () => {
        const wrapper = shallow(<RecoverWalletScreenComponent {...props} />);

        Mnemonic.verify = jest.fn().mockReturnValue(true);

        wrapper.setState({ mnemonic: validMnemonic.split(' ') });
        wrapper.find('[testID="button-confirm"]').simulate('press');

        const createWalletSpy = jest.spyOn(props, 'createHDWallet');

        expect(createWalletSpy.mock.calls).toMatchSnapshot();
        expect(props.navigation.navigate).toHaveBeenCalledWith(
            'MainNavigation',
            {},
            'navigate-action'
        );
    });
});
