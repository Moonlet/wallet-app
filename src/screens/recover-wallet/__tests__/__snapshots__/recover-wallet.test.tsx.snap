// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creat wallet terms screen component creates wallet and navigates to dashboard 1`] = `
Array [
  Array [
    "panic club above clarify orbit resist illegal feel bus remember aspect field test bubble dog trap awesome hand room rice heavy idle faint salmon",
    [Function],
  ],
]
`;

exports[`creat wallet terms screen component fills first input corectly 1`] = `ShallowWrapper {}`;

exports[`creat wallet terms screen component marks errors 1`] = `ShallowWrapper {}`;

exports[`creat wallet terms screen component renders correctly 1`] = `
"<View style={{...}}>
  <View style={{...}}>
    <View style={{...}}>
      <View style={{...}}>
        <View style={{...}}>
          <withTheme(TextComponent) small={true} style={{...}}>
            1.
          </withTheme(TextComponent)>
          <TextInput testID=\\"input-0\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" onChangeText={[Function: onChangeText]} onFocus={[Function: onFocus]} value=\\"\\" allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
        </View>
        <View style={{...}}>
          <withTheme(TextComponent) small={true} style={{...}}>
            2.
          </withTheme(TextComponent)>
          <TextInput testID=\\"input-1\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" onChangeText={[Function: onChangeText]} onFocus={[Function: onFocus]} value=\\"\\" allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
        </View>
        <View style={{...}}>
          <withTheme(TextComponent) small={true} style={{...}}>
            3.
          </withTheme(TextComponent)>
          <TextInput testID=\\"input-2\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" onChangeText={[Function: onChangeText]} onFocus={[Function: onFocus]} value=\\"\\" allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
        </View>
        <View style={{...}}>
          <withTheme(TextComponent) small={true} style={{...}}>
            4.
          </withTheme(TextComponent)>
          <TextInput testID=\\"input-3\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" onChangeText={[Function: onChangeText]} onFocus={[Function: onFocus]} value=\\"\\" allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
        </View>
      </View>
      <View style={{...}}>
        <View style={{...}}>
          <withTheme(TextComponent) small={true} style={{...}}>
            5.
          </withTheme(TextComponent)>
          <TextInput testID=\\"input-4\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" onChangeText={[Function: onChangeText]} onFocus={[Function: onFocus]} value=\\"\\" allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
        </View>
        <View style={{...}}>
          <withTheme(TextComponent) small={true} style={{...}}>
            6.
          </withTheme(TextComponent)>
          <TextInput testID=\\"input-5\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" onChangeText={[Function: onChangeText]} onFocus={[Function: onFocus]} value=\\"\\" allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
        </View>
        <View style={{...}}>
          <withTheme(TextComponent) small={true} style={{...}}>
            7.
          </withTheme(TextComponent)>
          <TextInput testID=\\"input-6\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" onChangeText={[Function: onChangeText]} onFocus={[Function: onFocus]} value=\\"\\" allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
        </View>
        <View style={{...}}>
          <withTheme(TextComponent) small={true} style={{...}}>
            8.
          </withTheme(TextComponent)>
          <TextInput testID=\\"input-7\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" onChangeText={[Function: onChangeText]} onFocus={[Function: onFocus]} value=\\"\\" allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
        </View>
      </View>
      <View style={{...}}>
        <View style={{...}}>
          <withTheme(TextComponent) small={true} style={{...}}>
            9.
          </withTheme(TextComponent)>
          <TextInput testID=\\"input-8\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" onChangeText={[Function: onChangeText]} onFocus={[Function: onFocus]} value=\\"\\" allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
        </View>
        <View style={{...}}>
          <withTheme(TextComponent) small={true} style={{...}}>
            10.
          </withTheme(TextComponent)>
          <TextInput testID=\\"input-9\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" onChangeText={[Function: onChangeText]} onFocus={[Function: onFocus]} value=\\"\\" allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
        </View>
        <View style={{...}}>
          <withTheme(TextComponent) small={true} style={{...}}>
            11.
          </withTheme(TextComponent)>
          <TextInput testID=\\"input-10\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" onChangeText={[Function: onChangeText]} onFocus={[Function: onFocus]} value=\\"\\" allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
        </View>
        <View style={{...}}>
          <withTheme(TextComponent) small={true} style={{...}}>
            12.
          </withTheme(TextComponent)>
          <TextInput testID=\\"input-11\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" onChangeText={[Function: onChangeText]} onFocus={[Function: onFocus]} value=\\"\\" allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
        </View>
      </View>
      <View style={{...}}>
        <View style={{...}}>
          <withTheme(TextComponent) small={true} style={{...}}>
            13.
          </withTheme(TextComponent)>
          <TextInput testID=\\"input-12\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" onChangeText={[Function: onChangeText]} onFocus={[Function: onFocus]} value=\\"\\" allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
        </View>
        <View style={{...}}>
          <withTheme(TextComponent) small={true} style={{...}}>
            14.
          </withTheme(TextComponent)>
          <TextInput testID=\\"input-13\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" onChangeText={[Function: onChangeText]} onFocus={[Function: onFocus]} value=\\"\\" allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
        </View>
        <View style={{...}}>
          <withTheme(TextComponent) small={true} style={{...}}>
            15.
          </withTheme(TextComponent)>
          <TextInput testID=\\"input-14\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" onChangeText={[Function: onChangeText]} onFocus={[Function: onFocus]} value=\\"\\" allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
        </View>
        <View style={{...}}>
          <withTheme(TextComponent) small={true} style={{...}}>
            16.
          </withTheme(TextComponent)>
          <TextInput testID=\\"input-15\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" onChangeText={[Function: onChangeText]} onFocus={[Function: onFocus]} value=\\"\\" allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
        </View>
      </View>
      <View style={{...}}>
        <View style={{...}}>
          <withTheme(TextComponent) small={true} style={{...}}>
            17.
          </withTheme(TextComponent)>
          <TextInput testID=\\"input-16\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" onChangeText={[Function: onChangeText]} onFocus={[Function: onFocus]} value=\\"\\" allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
        </View>
        <View style={{...}}>
          <withTheme(TextComponent) small={true} style={{...}}>
            18.
          </withTheme(TextComponent)>
          <TextInput testID=\\"input-17\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" onChangeText={[Function: onChangeText]} onFocus={[Function: onFocus]} value=\\"\\" allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
        </View>
        <View style={{...}}>
          <withTheme(TextComponent) small={true} style={{...}}>
            19.
          </withTheme(TextComponent)>
          <TextInput testID=\\"input-18\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" onChangeText={[Function: onChangeText]} onFocus={[Function: onFocus]} value=\\"\\" allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
        </View>
        <View style={{...}}>
          <withTheme(TextComponent) small={true} style={{...}}>
            20.
          </withTheme(TextComponent)>
          <TextInput testID=\\"input-19\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" onChangeText={[Function: onChangeText]} onFocus={[Function: onFocus]} value=\\"\\" allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
        </View>
      </View>
      <View style={{...}}>
        <View style={{...}}>
          <withTheme(TextComponent) small={true} style={{...}}>
            21.
          </withTheme(TextComponent)>
          <TextInput testID=\\"input-20\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" onChangeText={[Function: onChangeText]} onFocus={[Function: onFocus]} value=\\"\\" allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
        </View>
        <View style={{...}}>
          <withTheme(TextComponent) small={true} style={{...}}>
            22.
          </withTheme(TextComponent)>
          <TextInput testID=\\"input-21\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" onChangeText={[Function: onChangeText]} onFocus={[Function: onFocus]} value=\\"\\" allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
        </View>
        <View style={{...}}>
          <withTheme(TextComponent) small={true} style={{...}}>
            23.
          </withTheme(TextComponent)>
          <TextInput testID=\\"input-22\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" onChangeText={[Function: onChangeText]} onFocus={[Function: onFocus]} value=\\"\\" allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
        </View>
        <View style={{...}}>
          <withTheme(TextComponent) small={true} style={{...}}>
            24.
          </withTheme(TextComponent)>
          <TextInput testID=\\"input-23\\" style={{...}} placeholderTextColor=\\"#9a9a9a\\" placeholder=\\"\\" autoCapitalize=\\"none\\" autoCorrect={false} selectionColor=\\"#00DAFF\\" onChangeText={[Function: onChangeText]} onFocus={[Function: onFocus]} value=\\"\\" allowFontScaling={true} rejectResponderTermination={true} underlineColorAndroid=\\"transparent\\" />
        </View>
      </View>
    </View>
    <ScrollViewMock horizontal={true} overScrollMode=\\"never\\" snapToAlignment=\\"start\\" showsHorizontalScrollIndicator={false} style={{...}}>
      <View style={{...}} />
      <View style={{...}} />
    </ScrollViewMock>
  </View>
  <View style={{...}}>
    <withTheme(ButtonComponent) testID=\\"button-paste\\" style={{...}} onPress={[Function: onPress]}>
      Paste
    </withTheme(ButtonComponent)>
    <withTheme(ButtonComponent) testID=\\"button-next\\" style={{...}} disabled={false} onPress={[Function: onPress]}>
      Next
    </withTheme(ButtonComponent)>
    <withTheme(ButtonComponent) testID=\\"button-confirm\\" style={{...}} primary={true} onPress={[Function: onPress]}>
      Confirm
    </withTheme(ButtonComponent)>
  </View>
</View>"
`;

exports[`creat wallet terms screen component sets correct navigation options 1`] = `
<withTheme(HeaderLeftComponent)
  icon="arrow-left-1"
  onPress={[Function]}
  text="Back"
/>
`;

exports[`creat wallet terms screen component sets correct navigation options 2`] = `
Object {
  "headerLeft": [Function],
  "title": "Recover",
}
`;
